### ISSUE TITLE
Use rimraf to delete the dist folder

### ISSUE THREAD

**Issue by bompi88** [sircharleswatson/Electrometeor#16]
I frequently got error while the `dist.js` script tried to delete the `./dist` folder on Windows. I added rimraf to package.json:

```json
...
"devDependencies": {
  // rest of the dependencies
  
  "rimraf": "^2.4.1"
}
...
```

and wrapped the code inside a rimraf callback like this:

```js
var rmdir = require('rimraf');
...

cd(base);

// Remove with rimraf
rmdir(base + '/dist', function(error){
  
  mkdir('./dist');

...
}
```

Seems like it fixed my problem.

**Comment by sircharleswatson** [comment]
Yes, I remember having issues with it not building because it wouldn't remove the old folder for some reason. I sort of ignored the problem and just manually deleted it every time. Does this solve that issue?

**Comment by bompi88** [comment]
The rimraf module tries to remove the folder several times if `EBUSY` or `ENOTEMPTY ` error occur. It removes the folders in most cases. I have not used it before, so I cannot guarantee that it is the best choice available, but it looks like it works pretty well.

**Comment by sircharleswatson** [comment]
Alright. if you continue to use it and you don't run into any problems, I'll consider adding that.

**Comment by sjmcdowall** [comment]
You could also just use the fs-extra module .. which has a nice removeSync() function added (and is fully backward compatible with "fs" module) ..

**Comment by bompi88** [comment]
Yes, I have actually switched to fs-extra. I think fs-extra uses the rimraf module beneath the cover, and it's more convenient to use as well.

**Comment by sircharleswatson** [comment]
@bompi88 Do you want to make a small PR to fix that?
